### PR TITLE
Add net.sourceforge.jpdftweak.jPdfTweak

### PR DESCRIPTION
**Pull request by felipehw**
This is a Flatpak for [jPdfTweak](http://jpdftweak.sourceforge.net/) (a Java application).
I've contacted the developer and he allows me to maintain this package.

### CODE REVIEW

**Comment by Nokia808** [comment]
@felipehw 
Hi. I'm very interested with this program that it's developer stop from it's developing .... Dear are you just maintainer or you can also adding new feature to it ? If you can adding new features, then please alow me to contact you on other canal ...

**Comment by felipehw** [comment]
Just a mantainer :(

File: net.sourceforge.jpdftweak.jPdfTweak.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  "runtime-version" : "19.08",
+  "sdk" : "org.freedesktop.Sdk",
+  "sdk-extensions" : [ "org.freedesktop.Sdk.Extension.openjdk11" ],
+  "command": "net.sourceforge.jpdftweak.jPdfTweak",
+  "finish-args" : [
+    "--env=PATH=/app/jre/bin:/usr/bin:/app/bin",
+    "--share=network",
+    "--share=ipc",
+    "--socket=x11",
+	"--socket=fallback-x11",
```
Comment (nedrichards) [inline net.sourceforge.jpdftweak.jPdfTweak.json:14]: you need to have either socket=x11 or fallback-x11, not both. Also, the indenting.

If this works perfectly under wayland then you likely want fallback-x11.

File: net.sourceforge.jpdftweak.jPdfTweak.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    "--socket=wayland",
+    "--socket=pulseaudio",
+    "--device=dri",
+    "--filesystem=host",
+    "--filesystem=xdg-run/dconf", "--filesystem=~/.config/dconf:ro",
```
Comment (nedrichards) [inline net.sourceforge.jpdftweak.jPdfTweak.json:19]: I believe you don't need this under 19.08, I may be incorrect though

File: net.sourceforge.jpdftweak.jPdfTweak.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  "finish-args" : [
+    "--env=PATH=/app/jre/bin:/usr/bin:/app/bin",
+    "--share=network",
+    "--share=ipc",
+    "--socket=x11",
+	"--socket=fallback-x11",
+    "--socket=wayland",
+    "--socket=pulseaudio",
+    "--device=dri",
+    "--filesystem=host",
```
Comment (nedrichards) [inline net.sourceforge.jpdftweak.jPdfTweak.json:18]: for what functionality does this need host permissions?

File: net.sourceforge.jpdftweak.jPdfTweak.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      ],
+      "sources": [{
+        "type": "file",
+        "path": "jpdftweak"
+      },{
+        "type": "file",
+        "path": "net.sourceforge.jpdftweak.jPdfTweak.desktop"
+      },{
+        "type": "file",
+        "path": "net.sourceforge.jpdftweak.jPdfTweak.appdata.xml"
```
Comment (LeandroStanger) [inline net.sourceforge.jpdftweak.jPdfTweak.json:60]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

File: net.sourceforge.jpdftweak.jPdfTweak.json [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    "--share=network",
+    "--share=ipc",
+    "--socket=x11",
+	"--socket=fallback-x11",
+    "--socket=wayland",
+    "--socket=pulseaudio",
+    "--device=dri",
+    "--filesystem=host",
+    "--filesystem=xdg-run/dconf", "--filesystem=~/.config/dconf:ro",
+    "--talk-name=ca.desrt.dconf", "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
```
Comment (LeandroStanger) [inline net.sourceforge.jpdftweak.jPdfTweak.json:20]:
```suggestion
```

File: net.sourceforge.jpdftweak.jPdfTweak.json (hunk 1 above)
Comment (felipehw) [reply net.sourceforge.jpdftweak.jPdfTweak.json]: I tested ... and I need `socket=x11`, so I removed "fallback-x11" and a few other permissions (like `network`).

File: net.sourceforge.jpdftweak.jPdfTweak.json (hunk 2 above)
Comment (felipehw) [reply net.sourceforge.jpdftweak.jPdfTweak.json]: I removed this part too. And changed the filesystem to `--filesystem=home`.

File: net.sourceforge.jpdftweak.jPdfTweak.json (hunk 3 above)
Comment (felipehw) [reply net.sourceforge.jpdftweak.jPdfTweak.json]: I changed the filesystem to `--filesystem=home`.

File: net.sourceforge.jpdftweak.jPdfTweak.json (hunk 4 above)
Comment (felipehw) [reply net.sourceforge.jpdftweak.jPdfTweak.json]: I changed the filename to `net.sourceforge.jpdftweak.jPdfTweak.metainfo.xml`

File: net.sourceforge.jpdftweak.jPdfTweak.json (hunk 5 above)
Comment (felipehw) [reply net.sourceforge.jpdftweak.jPdfTweak.json]: OK! Removed!

File: net.sourceforge.jpdftweak.jPdfTweak.json (hunk 1 above)
Comment (nedrichards) [reply net.sourceforge.jpdftweak.jPdfTweak.json]: Do you mean it doesn't work at all under Wayland? If so you need to remove that permission, wayland users should fall back to XWayland fine.

Comment (felipehw) [reply net.sourceforge.jpdftweak.jPdfTweak.json]: This app doesn't work under Wayland. So I removed the Wayland permission. Now, this app only has `"--socket=x11"`.

File: net.sourceforge.jpdftweak.jPdfTweak.json [hunk 6]
```diff
@@ -0,0 +1,58 @@
+{
+	"app-id" : "net.sourceforge.jpdftweak.jPdfTweak",
+	"branch" : "stable",
```
Comment (bilelmoussaoui) [inline net.sourceforge.jpdftweak.jPdfTweak.json:3]:
```suggestion
```
it's already the default on Flathub

**Comment by bilelmoussaoui** [comment]
bot, build net.sourceforge.jpdftweak.jPdfTweak

**Comment by felipehw** [comment]
@bilelmoussaoui 
> bot, build net.sourceforge.jpdftweak.jPdfTweak

Concerning *Build 19038 failed* ...

Should I change the URL of the screenshot to another domain while the repo (`https://github.com/flathub/net.sourceforge.jpdftweak.jPdfTweak`) isn't created?

```
builddir/files/share/appdata/net.sourceforge.jpdftweak.jPdfTweak.appdata.xml: FAILED:
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://raw.githubusercontent.com/flathub/net.sourceforge.jpdftweak.jPdfTweak/master/img/screenshot-01.png]
```

File: net.sourceforge.jpdftweak.jPdfTweak.json (hunk 6 above)
Comment (felipehw) [reply net.sourceforge.jpdftweak.jPdfTweak.json]: Removed.

**Comment by nedrichards** [comment]
@felipehw yeah, you can put it in a temporary location so we can get a test build and verify everything. Just revert the commit when accepted.

File: net.sourceforge.jpdftweak.jPdfTweak.json [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+			"buildsystem": "simple",
+			"build-commands": [
+				"unzip jpdftweak-dist.zip -d dist",
+				"mkdir -p /app/jpdftweak",
+				"cp -r dist/* /app/jpdftweak",
+				"for size in 16 32 48 64 128 256; do install -Dm644 /app/jpdftweak/icons/icon${size}.png /app/share/icons/hicolor/${size}x${size}/apps/net.sourceforge.jpdftweak.jPdfTweak.png; done"
+			],
+			"sources": [{
+				"type": "file",
+				"url": "http://prdownloads.sourceforge.net/jpdftweak/jpdftweak-1.1.zip?download",
```
Comment (barthalion) [inline net.sourceforge.jpdftweak.jPdfTweak.json:31]: This is odd. There's no other way to have a download url which does not contain a query string?

Comment (felipehw) [reply net.sourceforge.jpdftweak.jPdfTweak.json]: Oooops. Query string removed!

**Comment by felipehw** [comment]
> @felipehw yeah, you can put it in a temporary location so we can get a test build and verify everything. Just revert the commit when accepted.

Updated the screenshot URL.

**Comment by bilelmoussaoui** [comment]
bot, build net.sourceforge.jpdftweak.jPdfTweak

**Pull request closed without merging**